packages/@aws-cdk/service-spec-importers/src/types/registry-schema/JsonSchema.ts

```diff
@@ -79,7 +83,7 @@ export namespace jsonschema {
   }
 
   export function isAnyOf(x: Schema): x is AnyOf<any> {
-    return !isAnyType(x) && 'anyOf' in x;
+    return !isAnyType(x) && !isTypeDefined(x) && 'anyOf' in x;
```

Comment:
Because of a transform that happens (somewhere in `patches/json-schema.ts` or something like that), I'm pretty sure this condition will never be true.

So `isAnyOf` will always return `false`, and we will have disabled the code paths that this tried to handle unions.

I think the explanation I posted to you on Slack will be useful for the future as well. Could you reformat that a bit and drop it into a Markdown file somewhere in the repo here?

Reply:
Buuuuttt... looking at the diff, this change definitely did something that looks like it works. Can you explain why it works? :)